CommonVCs/CategoryVC.swift

```diff
@@ -0,0 +1,75 @@
+//
+//  CategoryVC.swift
+//  To-Do App
+//
+//  Created by Ananth Desai on 18/02/22.
+//
+
+import UIKit
+
+class CategoryVC: UIViewController, UITableViewDelegate, UITableViewDataSource {
+    
+    var selectedIndex: Int = 0
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return sourceData.count
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = returnTableView.dequeueReusableCell(withIdentifier: "Cell", for: indexPath)
+        cell.textLabel?.text = sourceData[indexPath.row].title
+        cell.backgroundColor = .systemGreen
+        cell.accessoryType = .disclosureIndicator
+        return cell
+    }
+    
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        let newVC = ItemsVC(indexPath.row)
+        selectedIndex = indexPath.row
+        newVC.title = sourceData[indexPath.row].title
+        navigationController?.pushViewController(newVC, animated: true)
+    }
+    
+    let returnTableView: UITableView = {
+        let table = UITableView()
+        table.register(UITableViewCell.self, forCellReuseIdentifier: "Cell")
+        return table
+    }()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Do any additional setup after loading the view
+        view.addSubview(returnTableView)
+        returnTableView.dataSource = self
+        returnTableView.delegate = self
+        
+        self.navigationItem.rightBarButtonItem = UIBarButtonItem(
+            title: "Add",
+            style: .plain,
+            target: self,
+            action: #selector(tappedAdd)
+        )
+    }
+
+    override func viewDidAppear(_ animated: Bool) {
+        returnTableView.reloadData()
+        print("Root View appeared")
+    }
+    
+    override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+        returnTableView.frame = view.bounds
+    }
+    
+    @objc func tappedAdd(vc: UIViewController) {
```

Comment:
function name is not right here. This is  handling a user interaction, so it should be read something like `onAddButtonTap`, `addButtonTapHandler`. 
`tappedAdd` name doesn't give any meaning.

Reply:
Also there is a parameter `vc` why is that required ?